internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1147,6 +1186,11 @@ func (r *BMACReconciler) newSpokeBMH(log logrus.FieldLogger, bmh *bmh_v1alpha1.B
 		if _, err := r.reconcileAgentInventory(log, bmhSpoke, agent).Result(); err != nil {
 			return err
 		}
+
+		// check if we need to reset the agent in case of unbinding after installation
+		if _, err := r.reconcileUnboundAgent(log, bmhSpoke, agent).Result(); err != nil {
+			return err
+		}
```

Comment:
Could you elaborate on why this is needed here? Asking because this function is called *only* for day2 hosts and it modifies the BMH CR in the *spoke* cluster. This is done by copying the hub's BMH data and creating a new CR so we can expect the spoke BMH to be the same as the hub's.